benchmarks/cpp/utils.cpp

```diff
@@ -190,14 +190,14 @@ int64_t runBenchmarkIterations(
               ->groups()
               .size() > 1;
 
-  const auto& compile_log = executor_cache->getMostRecentExecutorInfo();
-  auto params = toString(compile_log.params);
-  auto lparams = toString(
-      compile_log.fusion_executor->as<KernelExecutor>()->lastLaunchParams());
   // Only set if not segmented. In the case of segmented fusions,
   // this could be confusing as the log would refect only the last
   // segment. Revisit if necessary.
   if (!segmented) {
+    const auto& compile_log = executor_cache->getMostRecentExecutorInfo();
+    auto params = toString(compile_log.params);
+    auto lparams = toString(
+        compile_log.fusion_executor->as<KernelExecutor>()->lastLaunchParams());
```

Comment:
only compute these when needed.

i.e. if segmentation happens with NoOp segment, the `toString` method above would fail.

Reply:
What do you mean by `if segmentation happens with NoOp segment`?

Reply:
If segmentation happened and gives no-op segments, since `toString` here isn't expecting no-op scheduler, we run into asserts.
https://github.com/NVIDIA/Fuser/blob/e286c6f85a535dd462a2913c671a4e76412f6ba8/benchmarks/cpp/utils.cpp#L131-L132